doc/configuration.mld

```diff
@@ -0,0 +1,35 @@
+{0 How OCamlFormat computes its configuration}
+
+{1 Configuration files}
+
+Ocamlformat fetches the following files on the FS:
+
++ [.git], [.hg] or [dune-project]
++ [.ocamlformat] and [.ocp-indent] files
++ [.ocamlformat-ignore] files
++ [.ocamlformat-enable] files
+
+(1.) files are used to determine the {b project root}, which can be overriden by the [--root] option, they are looked up from the path of the file to format, and upwards following parent directories, until the first one is found.
+
+(2.) (3.) and (4.) are looked up from the path of the file to format, and following up the parents up to the project root.
+
+(2.) are the {b configuration files}, they  contain the options used to configure ocamlformat.
+
+(3.) contain a list of files to exclude from the formatting, if the formatting is enabled ([disable = false]).
+
+(4.) contain a list of files to include in the formatting, if the formatting is disabled ([disable = true]).
```

Comment:
I think we can drop this and go directly to the next section as it doesn't say much and the next section is much more interesting.